### ISSUE TITLE
Multiple Chromecasts not reliably detected

### ISSUE THREAD

**Issue by danpowpow** [d8ahazard/Phlex#122]
Phlex is pretty darn good :-)
Unfortunately it doesn't reliably see all my chromecasts, and simply times out when I try to cast to a devict that isn't discovered.

Strangely,  the Discovered CCs change each scan, so whilst I may have 7 or 8 showing at any one time, they will be different ones... 

I am running phlex using docker on OMV2. No apparent network/connectivity issues other than this.

Is there a limit on max CCs scannable? I have 7 CCaudio, 2CCvideo, and 6 CCaudio groups.

**Comment by philastrophist** [comment]
How did you get Phlex to detect any chromecasts? I can't seem to make it work.

I've uncommented `extension=php_sockets.dll` but that doesn't seem to help. Everything is on the same network, so it should find them like plex-web does, right?

**Comment by danpowpow** [comment]
Phlex discovered my chromecasts without me commenting/uncommenting anything.  i simply enabled the switch in the gui - settings>plex>use cast devices.
Good luck
@d8ahazard  - any ideas on my initial problem?

**Comment by clambert11** [comment]
Running be3203f in Docker on a Synology DS916+.  I'm also struggling to detect any of my 4 Chromecasts,  I'm behind Google Wi-Fi for what it's worth.

**Comment by JonBlackwell** [comment]
I am also having this issue. I have 4 CC's (audio), 3 CC's (video, 1 an ultra), 2 Google Home's, and 4 Groups. Phlex only discovers the 4 audio CC's, 1 video CC (the ultra), and 1 group.

**Comment by clambert11** [comment]
I'm not sure if this helps but I noticed Plex on Android itself does something similar for me.  When I hit cast it doesn't find the Chromecasts initially but does see some other devices (Plex running on various machines).  If I then refresh in the app to locate more devices it then does find the other Chromecasts.  Could Phlex be doing something similar?  Unfortunately hitting refresh in Phlex doesn't help find them.

**Comment by monsieurdayfrank** [comment]
I am also experiencing issues with CC's not being detected. I am running window 7 64-bit. I installed XAMPP and run the apache server. I downloaded the zip from github, unpacked it and put it in htdocs. all configured, apache starts, i can hit the phlex website, link my google account and can talk to phlex via google home.

the first night, only my galaxy S8+ and my son's galaxy Luna showed up as available devices from the phlex page. and i could ask phlex to cast to those devices and it worked. i am really only interested in CC's though. 

So, I tried some of the tweaks mentioned above (timeout of 90 instead of 15) but that's only completely removed all devices from the phlex device list. the device list is now completely empty. i've re-dropped the phlex directory from the downloaded zip file so the files are all unmodified.

all devices/computers/servers are on the same home network. wireless devices are all on the same wifi or wireless AP. all 5 CC's are MIA. :/

**Comment by d8ahazard** [comment]
Okay, so...basically...cast discovery in PHP is tricky.  There is no official cast library for PHP, and the MDNS one has never worked for me.  So, everything Phlex uses is based on a lot of trial and error, and @ChrisRidings doing everything he can given the devices he has available.  

So, for the issue of "it doesn't find them all, all the time", I'm working on a cludgy, but surefire solution...letting you hardcode them in.  Basically, you'll be able to manually add devices in Settings, and these will be appended/supersede those picked up by a scan.

Obv, this is not a trivial task, but I've got the groundwork laid for it in the UI and api, I just need to work out some of the display elements and backend stuff, then I'll enable it.

@clambert11 - To test for sure, I'd suggest plugging a PC/laptop into the same LAN cable/switch/port you use for your Syno - then load up the Plex web app.  If you see cast devices there, then we know it's not a routing issue.  

Either way, I'd like to see the logs from you doing a device discovery, make sure "obscure sensitive data" is enabled in settings, then do a rescan and copypaste the relevant lines.

@philastrophist Is "Use cast devices" enable in Settings -> Plex?

**Comment by clambert11** [comment]
@d8ahazard I sent the logs in a private message in the Plex forums after scanning for devices.  Even though I had "obscure sensitive data" on there was some other info I didn't want out there.

I've got 4 Chromecasts in the house.  Two are on at all times.  The logs were generated on a laptop plugged into the same switch as my DS916+.  I had Plex open in a web browser at the same time on my Chromebook.  Phlex is able to locate this.  Oddly enough, the Devices tab was on the laptop plugged into the switch.  It wouldn't let me manually add anything though.  When I used my Chromebook to connect to the same Docker container the Devices tab was not there.  It kinda blew my mind.
